packages/core/src/events/input/load/vectorstore/baseretriever.ts

```diff
@@ -0,0 +1,54 @@
+
+import { Context } from "../docs/context.js";
+import { Callable } from "../../../../record/callable.js";
+import { CallableConfig } from "../../../../record/callable.js";
+
+
+export interface BaseRetrieverInput {
+    callbacks?: any;
+    tags?: string[];
+    metadata?: Record<string, unknown>;
+    verbose?: boolean;
+}
+
+
+export abstract class BaseRetriever extends Callable<string, Context[]> {
+  callbacks?: any;
+
+  tags?: string[];
+
+  metadata?: Record<string, unknown>;
+
+  verbose?: boolean;
+
+  constructor(fields?: BaseRetrieverInput) {
+    super(fields);
+    this.callbacks = fields?.callbacks;
+    this.tags = fields?.tags ?? [];
+    this.metadata = fields?.metadata ?? {};
+    this.verbose = fields?.verbose ?? false;
```

Comment:
Probably don't need to init these variables.

Reply:
got it, will remove